spec.md

```diff
@@ -1096,6 +1010,30 @@ Ports are always considered from the perspective of "inside the module".
 Moreover, input ports may only appear "on the left side" of a connect, while output ports may only appear "on the right side" of a connect.
 Finally, while submodules instances and memories are strictly sources, they interact with the sub-field rule below, allowing connections to their input ports
 
+## Type Equivalence
+
+The type equivalence relation determines whether one values of one type can be coerced into values of another.
+
+An unsigned integer type is always equivalent to another unsigned integer type regardless of bit width, and is not equivalent to any other type.
+Similarly, a signed integer type is always equivalent to another signed integer type regardless of bit width, and is not equivalent to any other type.
+
+Clock types are equivalent to clock types, and are not equivalent to any other type.
+
+An uninferred `Reset`{.firrtl} can be connected to another `Reset`{.firrtl}, `UInt`{.firrtl} of unknown width, `UInt<1>`{.firrtl}, or `AsyncReset`{.firrtl}.
+It cannot be connected to both a `UInt`{.firrtl} and an `AsyncReset`{.firrtl}.
```

Comment:
This sentence doesn't seem appropriate for type equivalence.

Reply:
This was the existing copy. It's in rough shape.

I want to make a separate PR at some point to rename "type equivalence" to something more sane.